src/app/core/services/local-storage.service.test.ts

```diff
@@ -0,0 +1,345 @@
+  backupsBucket: 'backups-bucket',
+  root_folder_id: 1,
+  rootFolderId: 'folder-id-123',
+  rootFolderUuid: 'folder-uuid-456',
+  sharedWorkspace: false,
+  credit: 100,
+  mnemonic: 'test mnemonic phrase',
+  privateKey: 'private-key-mock',
+  publicKey: 'public-key-mock',
+  revocationKey: 'revocation-key-mock',
+  keys: {
+    ecc: {
+      publicKey: 'ecc-public-key-mock',
+      privateKey: 'ecc-private-key-mock',
+    },
+    kyber: {
+      publicKey: 'kyber-public-key-mock',
+      privateKey: 'kyber-private-key-mock',
+    },
+  },
+  teams: true,
+  appSumoDetails: null,
+  registerCompleted: true,
+  hasReferralsProgram: true,
+  createdAt: new Date('2023-06-01T12:00:00.000Z'),
+  avatar: null,
+  emailVerified: true,
+};
+
+const mockWorkspaceCredentialsDetails: WorkspaceCredentialsDetails = {
+  workspaceId: 'workspace-123',
+  bucket: 'workspace-bucket',
+  workspaceUserId: 'workspace-user-456',
+  email: 'workspace.user@example.com',
+  credentials: {
+    networkPass: 'mockNetworkPassword123',
+    networkUser: 'workspace.network.user',
+  },
+  tokenHeader: 'Bearer mock-token-abc-123',
+};
+
+const mockWorkspaceData: WorkspaceData = {
+  workspaceUser: {
+    backupsUsage: '500000000', // 500 MB
+    createdAt: '2023-05-01T10:00:00.000Z',
+    deactivated: false,
+    driveUsage: '1200000000', // 1.2 GB
+    freeSpace: '8800000000', // 8.8 GB free
+    id: 'workspace-user-001',
+    isManager: true,
+    isOwner: false,
+    key: 'mock-encryption-key',
+    member: {
+      avatar: null,
+      backupsBucket: 'backups-bucket-id',
+      bridgeUser: 'bridge-user-mock',
+      credit: 100,
+      email: 'jane.doe@example.com',
+      errorLoginCount: 0,
+      id: 101,
+      isEmailActivitySended: true,
+      lastPasswordChangedAt: '2023-07-01T12:00:00.000Z',
+      lastResend: '2023-08-01T12:00:00.000Z',
+      lastname: 'Doe',
+      name: 'Jane',
+      referralCode: 'REF-JANE-123',
+      referrer: null,
+      registerCompleted: true,
+      rootFolderId: 2001,
+      sharedWorkspace: true,
+      syncDate: '2023-10-01T08:00:00.000Z',
+      userId: 'user-1234',
+      username: 'janedoe',
+      uuid: 'uuid-janedoe-5678',
+      welcomePack: true,
+    },
+    memberId: '101',
+    rootFolderId: 'folder-abc-123',
+    spaceLimit: '10000000000', // 10 GB
+    updatedAt: '2023-10-01T12:00:00.000Z',
+    usedSpace: '1700000000', // 1.7 GB
+    workspaceId: 'workspace-xyz-456',
+  },
+  workspace: {
+    id: 'workspace-xyz-456',
+    ownerId: 'user-1234',
+    address: '123 Main St, Example City',
+    name: 'Marketing Workspace',
+    description: 'Workspace for the marketing department',
+    defaultTeamId: 'team-789',
+    workspaceUserId: 'workspace-user-001',
+    setupCompleted: true,
+    createdAt: '2023-04-01T09:30:00.000Z',
+    updatedAt: '2023-10-01T12:30:00.000Z',
+    avatar: null,
+    rootFolderId: 'folder-abc-123',
+    phoneNumber: null,
+  },
+};
+
+describe('Testing the local storage service', () => {
```

Comment:
Maybe we can change these tests, instead of mock every return and spy every function, we could mock the local storage and test the `localStorageService` without mock every getItem, that way I think we could better simulate the `localstorageService` behaviour.
If it's ok with you, let's go through the tests and see if it makes sense :)